src/mainwindow.cpp

```diff
@@ -306,8 +308,11 @@ void MainWindow::handleDownload(QFile *file, QString filename, QVariant extra) {
     ModelMeta meta = extra.value<ModelMeta>();
     meta.installedOn = QDateTime::currentDateTimeUtc();
     auto model = models_.writeModel(file, meta, filename);
-    if (model) // if writeModel didn't fail
+    if (model) { // if writeModel didn't fail
         settings_.translationModel.setValue(model->path, Setting::AlwaysEmit);
+        models_.updateAvailableModels(); // @TODO This doesn't remove the newly downloaded model from the the list of
+                                         // of available models for downloading at this time. Need to fix it.
+    }
```

Comment:
Weird… `writeModel()` already calls `updateAvailableModels()` at the end.

Reply:
Yeah i noticed that, but even in master it doesn't work. Not exactly sure what goes wrong in there.

Reply:
I've made #114 for it because it looks like a logic bug somewhere (I suspect the downloaded local model is somehow slightly different from the scanned-at-startup local model and so the remote-local match-up fails?)